core/src/main/java/org/aerogear/mobile/core/http/HttpResponse.java

```diff
@@ -21,6 +21,8 @@
     HttpResponse onComplete(Runnable runnable);
 
     int getStatus();
+    boolean isFailed();
+    Exception getRequestError();
```

Comment:
Request errors should be sent asynchronously and processed then.  I don't think it is correct to have them also be properties of the response.

Reply:
I think i added this because there was no way to know when a request has failed and trying to use `getStatus` on a failed requests results in a NPE. Maybe i was using the wrapper wrong.